apps/desktop/src-tauri/src/inference/server.rs

```diff
@@ -58,24 +58,55 @@ async fn post_completions(payload: Json<CompletionRequest>) -> impl Responder {
 
   let (token_sender, receiver) = flume::unbounded::<Bytes>();
 
-  HttpResponse::Ok()
-    .append_header(("Content-Type", "text/event-stream"))
-    .append_header(("Cache-Control", "no-cache"))
-    .keep_alive()
-    .streaming({
-      let abort_flag = Arc::new(RwLock::new(false));
-
-      AbortStream::new(
-        receiver,
-        abort_flag.clone(),
-        start(InferenceThreadRequest {
-          model_guard: model_guard.clone(),
-          abort_flag: abort_flag.clone(),
-          token_sender,
-          completion_request: payload.0,
-        }),
-      )
-    })
+  if let Some(true) = payload.stream {
+    HttpResponse::Ok()
+      .append_header(("Content-Type", "text/event-stream"))
+      .append_header(("Cache-Control", "no-cache"))
+      .keep_alive()
+      .streaming({
+        let abort_flag = Arc::new(RwLock::new(false));
+        let str_buffer = Arc::new(Mutex::new(String::new()));
+
+        AbortStream::new(
+          receiver,
+          abort_flag.clone(),
+          start(InferenceThreadRequest {
+            model_guard: model_guard.clone(),
+            abort_flag: abort_flag.clone(),
+            token_sender,
+            completion_request: payload.0,
+            nonstream_completion_tokens: str_buffer.clone(),
+            stream: true,
+            tx: None,
+          }),
+        )
+      })
+  } else {
+    let abort_flag = Arc::new(RwLock::new(false));
+    let completion_tokens = Arc::new(Mutex::new(String::new()));
+    let (tx, rx) = flume::unbounded::<()>();
+    start(InferenceThreadRequest {
+      model_guard: model_guard.clone(),
+      abort_flag: abort_flag.clone(),
+      token_sender,
+      completion_request: payload.0,
+      nonstream_completion_tokens: completion_tokens.clone(),
+      stream: false,
+      tx: Some(tx),
+    });
+
+    rx.recv().unwrap();
+
+    let locked_str_buffer = completion_tokens.lock();
+    let completion_body = json!({
+      "completion": locked_str_buffer.clone()
+    });
+
+    HttpResponse::Ok()
+      .append_header(("Content-Type", "text/plain"))
```

Comment:
We should return a response similar to openAI shape IMO, which is an `application/json` type. This way we can maximize the server's response shape compatibility with other client libs :-?

Reply:
I think that makes sense, yes. Will fix those. Thanks for looking at my code.